index.js

```diff
@@ -292,21 +314,38 @@ module.exports.shell = (cmd, opts) => handleShell(module.exports, cmd, opts);
 
 module.exports.sync = (cmd, args, opts) => {
 	const parsed = handleArgs(cmd, args, opts);
+	const joinedCmd = joinCmd(cmd, args);
 
 	if (isStream(parsed.opts.input)) {
 		throw new TypeError('The `input` option cannot be a stream in sync mode');
 	}
 
 	const result = childProcess.spawnSync(parsed.cmd, parsed.args, parsed.opts);
+	result.code = result.status;
 
-	if (result.error || result.status !== 0) {
-		throw (result.error || new Error(result.stderr === '' ? result.stdout : result.stderr));
-	}
+	if (result.error || result.status !== 0 || result.signal !== null) {
+		const err = makeError(result, {
+			joinedCmd,
+			parsed
+		});
+
```

Comment:
The async methods have one more property though, the `killed` property.

https://github.com/sindresorhus/execa/blob/master/index.js#L240-L243   

I'm not sure what to do with it because it doesn't really look like it is set or something. Any ideas on this one?

Reply:
That's because `killed` is part of the sub-process object, but the sync methods only return the result, not the sub-process:

<img width="1029" alt="screen shot 2017-09-29 at 13 19 38" src="https://user-images.githubusercontent.com/170270/31003255-ee7e73e6-a518-11e7-9f41-76db4d56df51.png">

It doesn't really make sense on the sync methods, so we should just document that's it's only available on the async methods.